pkg/ifaces/watcher.go

```diff
@@ -191,9 +199,19 @@ func (w *Watcher) netnsNotify(ctx context.Context, out chan Event) {
 				}
 				if event.Op&fsnotify.Create == fsnotify.Create {
 					ns := filepath.Base(event.Name)
-					log.WithField("netns", ns).Debug("netns notification")
+					log.WithField("netns", ns).Debug("netns create notification")
+					if _, ok := w.nsDone[ns]; ok {
+						log.WithField("netns", ns).Debug("netns channel already exists, delete it")
+						delete(w.nsDone, ns)
+					}
+					w.nsDone[ns] = make(chan struct{})
 					go w.sendUpdates(ctx, ns, out)
 				}
+				if event.Op&fsnotify.Remove == fsnotify.Remove {
+					ns := filepath.Base(event.Name)
+					log.WithField("netns", ns).Debug("netns delete notification")
+					w.nsDone[ns] <- struct{}{}
```

Comment:
there is a risk that this code is called after the channel was closed and deleted?
ie. `w.sendUpdates` finishes, and then we get a fsnotify.Remove?
(just asking)

Reply:
the code in sendUpdates is blocking won't finish till done is happen which is triggered via fsnotify

Reply:
I added a check around the remove to be safe